File: blablamovebackend/src/main/java/fr/polytech/unice/blablamove/teamc/blablamovebackend/model/influxdb/DeliveredItem.java
package fr.polytech.unice.blablamove.teamc.blablamovebackend.model.influxdb;

import org.influxdb.annotation.Column;
import org.influxdb.annotation.Measurement;

import java.time.Instant;

/**
 * Class DeliveredItem
 *
 * @author Tanguy Invernizzi
 */
@Measurement(name = "delivery_item")
public class DeliveredItem {

    /**
     * The type of the item.
     */
    @Column(name = "item_type")
    private String itemType;

    /**
     * The id of this delivery.
     */
    @Column(name = "delivery_uuid")
    private String delivery_uuid;

    /**
     * The time at which the issue happened.
     */
    @Column(name = "time")
    private Instant time;

    public Instant getTime() {
        return time;
    }

    public String getItemType() {
        return itemType;
    }

    public void setItemType(String itemType) {
        this.itemType = itemType;
    }

    public String getDelivery_uuid() {
        return delivery_uuid;
    }

    public void setDelivery_uuid(String delivery_uuid) {
        this.delivery_uuid = delivery_uuid;
    }

    public void setTime(Instant time) {
        this.time = time;
    }
}

File: adminMarketingUI/src/app/model/delivered-item.ts
export class DeliveredItem {
  private _item_type: string;
  private _delivery_uuid: string;
  private _time: Date;

  constructor(item_type: string, delivery_uuid: string, time: Date) {
    this._item_type = item_type;
    this._delivery_uuid = delivery_uuid;
    this._time = time;
  }

  get item_type(): string {
    return this._item_type;
  }

  set item_type(value: string) {
    this._item_type = value;
  }

  get delivery_uuid(): string {
    return this._delivery_uuid;
  }

  set delivery_uuid(value: string) {
    this._delivery_uuid = value;
  }

  get time(): Date {
    return this._time;
  }

  set time(value: Date) {
    this._time = value;
  }
}
